avro_schema/frontend.lua

```diff
@@ -978,9 +942,48 @@ function debug_print_indent(msg)
     print(msg)
 end
 
+--
+-- Compiler generates code from the IR, which is more generic than
+-- a bare schema. IR defines a 'from' schema, a 'to' schema and a mapping.
+--
+-- IR is a graph of nodes mirroring a schema structure.
+-- Simple terminal nodes (string):
+--
+--   NUL, BOOL, INT, LONG, FLT, DBL, BIN, STR,
+--   INT2LONG, INT2FLT, INT2DBL, LONG2FLT, LONG2DBL,
+--   FLT2DBL, STR2BIN, BIN2STR
+--
+-- Complex nodes (table):
+--
+--   { type = 'FIXED',  size   = <n>  , nullable = nil/true }
+--   { type = 'ARRAY',  nested = <ir> , nullable = nil/true }
+--   { type = 'MAP',    nested = <ir> , nullable = nil/true }
+--
+--   { type = 'ENUM',   nullable = nil/true,
+--     from = <schema>, to = <schema>, i2o = ? }
+--
+--   { type = 'RECORD', nullable = nil/true, nested = {
+--           type = '__RECORD__',
+--           from = <schema>, to = <schema>, i2o = ?, o2i = ?, ... }}
+--
+--   { type = 'UNION', nested = {
+--           type = '__UNION__',
+--           from = <schema>, to = <schema>, i2o = ?, ... }}
+--
+-- Legend: n      — size of fixed
```

Comment:
You have used em dash here, but ascii dash below. It worth to restrict ourself to ascii within a source code.

Reply:
Fixed.